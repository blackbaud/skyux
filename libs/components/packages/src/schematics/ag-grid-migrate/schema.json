{
  "$schema": "http://json-schema.org/schema",
  "type": "object",
  "properties": {
    "sourceRoot": {
      "type": "string",
      "description": "The path to the source files.",
      "default": "./"
    }
  },
  "required": ["sourceRoot"]
}
