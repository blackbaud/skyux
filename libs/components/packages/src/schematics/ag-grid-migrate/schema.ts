/**
 * Represents options passed via Angular CLI.
 */
export interface Schema {
  /**
   * Path to the source root of the project. Defaults to the current directory.
   */
  sourceRoot?: string;
  /**
   * The version of AG Grid to migrate from. Defaults to the version found in the project's package-lock.json.
   */
  from?: string;
}
