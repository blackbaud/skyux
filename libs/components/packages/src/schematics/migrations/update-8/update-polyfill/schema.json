{
  "$schema": "http://json-schema.org/schema",
  "$id": "UpdatePolyfill",
  "title": "Update polyfill to switch to config",
  "description": "Remove code from polyfills.ts/test.ts and add @skyux/packages/polyfills to workspace config.",
  "type": "object"
}
