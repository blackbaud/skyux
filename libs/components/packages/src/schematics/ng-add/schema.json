{
  "$schema": "http://json-schema.org/schema",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    }
  }
}
