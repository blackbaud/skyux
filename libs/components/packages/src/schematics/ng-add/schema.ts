/**
 * Represents options passed via Angular CLI.
 */
export interface Schema {
  /**
   * The name of the project.
   */
  project?: string;
}
