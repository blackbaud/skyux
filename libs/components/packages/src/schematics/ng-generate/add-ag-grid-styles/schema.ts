/**
 * Represents options passed via Angular CLI.
 */
export interface Schema {
  /**
   * The name of the project to add polyfills to.
   */
  project?: string;
}
