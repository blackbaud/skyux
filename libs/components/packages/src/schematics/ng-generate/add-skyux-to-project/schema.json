{
  "$schema": "http://json-schema.org/schema",
  "type": "object",
  "properties": {
    "project": {
      "type": "string"
    }
  },
  "required": ["project"]
}
