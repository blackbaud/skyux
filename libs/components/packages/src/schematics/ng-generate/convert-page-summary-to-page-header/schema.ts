/**
 * Represents options passed via Angular CLI.
 */
export interface Schema {
  /**
   * The name of the project to convert.
   */
  project?: string;
}
