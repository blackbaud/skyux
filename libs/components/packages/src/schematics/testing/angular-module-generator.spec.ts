import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { RoutingScope } from '@schematics/angular/module/schema';

import {
  angularComponentGenerator,
  angularModuleGenerator,
} from './angular-module-generator';
import { createTestApp } from './scaffold';

describe('angularModuleGenerator', () => {
  const runner = new SchematicTestRunner(
    'schematics',
    require.resolve('../../../collection.json'),
  );

  it('should generate an angular module', async () => {
    const tree = await createTestApp(runner, {
      projectName: 'my-app',
    });
    await angularModuleGenerator(runner, tree, {
      name: 'my-module',
      project: 'my-app',
    });
    expect(
      tree.readText('/src/app/my-module/my-module.module.ts'),
    ).toBeTruthy();
  });

  it('should generate an angular module with routing', async () => {
    const tree = await createTestApp(runner, {
      projectName: 'my-app',
    });
    await angularModuleGenerator(runner, tree, {
      name: 'my-module',
      project: 'my-app',
      routing: true,
      routingScope: RoutingScope.Root,
    });
    await angularModuleGenerator(runner, tree, {
      name: 'my-sub-module',
      project: 'my-app',
      module: 'my-module',
      route: 'my-sub-module',
      routing: true,
      routingScope: RoutingScope.Child,
    });
    expect(
      tree.readText('/src/app/my-module/my-module.module.ts'),
    ).toMatchSnapshot();
    expect(
      tree.readText('/src/app/my-sub-module/my-sub-module.module.ts'),
    ).toMatchSnapshot();
  });

  it('should generate an angular component', async () => {
    const tree = await createTestApp(runner, {
      projectName: 'my-app',
    });
    await angularComponentGenerator(runner, tree, {
      name: 'my-component',
      project: 'my-app',
      skipImport: true,
    });
    expect(
      tree.readText('/src/app/my-component/my-component.component.ts'),
    ).toBeTruthy();
  });
});
