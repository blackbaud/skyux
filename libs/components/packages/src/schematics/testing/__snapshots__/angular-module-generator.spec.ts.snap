// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`angularModuleGenerator should generate an angular module with routing 1`] = `
"import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { MyModuleRoutingModule } from './my-module-routing.module';


@NgModule({
  declarations: [],
  imports: [
    CommonModule,
    MyModuleRoutingModule
  ]
})
export class MyModuleModule { }
"
`;

exports[`angularModuleGenerator should generate an angular module with routing 2`] = `
"import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { MySubModuleRoutingModule } from './my-sub-module-routing.module';
import { MySubModuleComponent } from './my-sub-module.component';


@NgModule({
  declarations: [
    MySubModuleComponent
  ],
  imports: [
    CommonModule,
    MySubModuleRoutingModule
  ]
})
export class MySubModuleModule { }
"
`;
